src/main/java/com/gmail/nossr50/events/skills/salvage/McMMOPlayerSalvageCheckEvent.java

```diff
@@ -0,0 +1,67 @@
+package com.gmail.nossr50.events.skills.salvage;
+
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.inventory.ItemStack;
+
+import com.gmail.nossr50.datatypes.skills.SkillType;
+import com.gmail.nossr50.events.skills.McMMOPlayerSkillEvent;
+
+/**
+ * Called just before a player salvages an item with mcMMO.
+ */
+public class McMMOPlayerSalvageCheckEvent extends McMMOPlayerSkillEvent implements Cancellable {
+    private int salvageableAmount;
+    private ItemStack salvageMaterial;
+    private ItemStack salvageResults;
+    private ItemStack enchantedBook;
+    private boolean cancelled;
+
+    public McMMOPlayerSalvageCheckEvent(Player player, int salvageableAmount, ItemStack salvageMaterial, ItemStack salvageResults, ItemStack enchantedBook) {
+        super(player, SkillType.SALVAGE);
+        this.salvageableAmount = salvageableAmount;
+        this.salvageMaterial = salvageMaterial;
+        this.salvageResults = salvageResults;
+        this.enchantedBook = enchantedBook;
+        this.cancelled = false;
+    }
+
+    /**
+     * @return The amount of materials you get back after salvaging.
+     */
+    public int getSalvageableAmount() {
+        return salvageableAmount;
+    }
+
+    /**
+     * @return The item that should get salvaged.
+     */
+    public ItemStack getSalvageMaterial() {
+        return salvageMaterial;
+    }
+
+    /**
+     * @return The results that should be dropped after salvaging.
+     */
+    public ItemStack getSalvageResults() {
+        return salvageResults;
+    }
+
+    /**
+     * @return The enchanted book that should drop after salvaging or null if no book should be dropped.
+     */
+    public ItemStack getEnchantedBook() {
+        return enchantedBook;
+    }
+
+    /** Following are required for Cancellable **/
+    @Override
+    public boolean isCancelled() {
+        return false;
```

Comment:
If this always returns false setCancelled is pretty dang useless....

Reply:
So first of all thanks for taking the time to have a look at this and doing a review :+1:
Looks like I was to much in a rush for getting this finished and actually forgot to propperly implement getCancelled() and setCancelled() which my IDE created for me. :face_palm:
So sorry for this inconvinience.